### PR TITLE
Add analytics

### PR DESCRIPTION
**Pull request by sahil21**
Functionalities added in the "Analytics" class:
1. Hit event (like find peaks)
2. Hit screen view (like mainwindow, samples widget)
3. Session duration

### CODE REVIEW

File: src/gui/mzroll/main.cpp [hunk 1]
```diff
@@ -55,8 +55,9 @@ int main(int argc, char *argv[])
 {
 
     QApplication app(argc, argv);
-    qApp->setOrganizationName("ElucidataInc");
-    qApp->setApplicationName("El-Maven");
+    qApp->setOrganizationName("Elucidata");
+    qApp->setApplicationName("El-MAVEN");
+    qApp->setApplicationVersion("v0.4");
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/main.cpp]: @sahil21 Do we have to set this manually before every release?

Comment (sahil21) [reply src/gui/mzroll/main.cpp]: @shubhra-agrawal Added a commit to resolve this.

File: src/gui/mzroll/analytics.cpp [hunk 2]
```diff
@@ -81,6 +81,26 @@ void Analytics::httpPost(QUrlQuery query) {
...
+{
+    QString settingsPath = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation);
+    settingsPath = QDir::cleanPath(settingsPath + QDir::separator() + "El-MAVEN_analytics.ini");
+
+    QSettings settings(settingsPath, QSettings::IniFormat);
+    QString clientID;
+    if (!settings.contains("analytics-cid"))
+    {
+        clientID = QUuid::createUuid().toString();
+        settings.setValue("analytics-cid", clientID);
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/analytics.cpp]: @sahil21 Is the .ini file overwritten every time I install a new version? or does a user have a unique client ID across versions? Just curious.

Comment (sahil21) [reply src/gui/mzroll/analytics.cpp]: @shubhra-agrawal Unique id across versions.

File: src/gui/mzroll/main.cpp [hunk 3]
```diff
@@ -55,8 +55,9 @@ int main(int argc, char *argv[])
 {
 
     QApplication app(argc, argv);
-    qApp->setOrganizationName("ElucidataInc");
-    qApp->setApplicationName("El-Maven");
+    qApp->setOrganizationName("Elucidata");
+    qApp->setApplicationName("El-MAVEN");
```
[the remark this answers is not in the document]
Comment (sahil21) [reply src/gui/mzroll/main.cpp]: @rish9511 Sure. I have made the changes

**Comment by sahil21** [comment]
@rish9511 
- Options dialog false screen view has been resolved.
- Regarding static class, let's do this in the next release.

**Pull request merged**